cmd/clusterctl/api/v1alpha3/provider_type.go

```diff
@@ -50,7 +50,7 @@ type Provider struct {
 	// WatchedNamespace indicates the namespace where the provider controller is watching.
 	// If empty the provider controller is watching for objects in all namespaces.
 	//
-	// Deprecated: in clusterctl v1alpha4 all the providers watch all the namespaces; this field will be removed in a future version of this API
+	// Deprecated: in Cluster API v1alpha4 and above all providers watch all the namespaces; this field will be removed in a future version of this API
```

Comment:
should this really say `Cluster API`? AFAIK clusterctl is still on v1alpha3

Reply:
This change is related to the CAPI contract version, not the clusterctl version which is why I updated this. AFAIU it's related to https://cluster-api.sigs.k8s.io/developer/providers/v0.3-to-v0.4.html#multi-tenancy

Reply:
Looks like in our current code we always set this field to an empty string and never read it (except in an equals func, but I didn't find a usage for the equals func). So basically it seems to be dead code today.

I have no idea if the watch behavior was rather coupled to the clusterctl v1alpha3 API or the v1alpha3 contract.

My guess is, at this point there was no difference between:
* clusterctl v1alpha3
* contract v1alpha3
* other API types v1alpha3

I think technically there is no such thing as "all providers in Cluster API v1alpha4 and above". Cluster API itself doesn't have a global version which applies to core and all providers. The only thing that comes close is the contract version.